src/test/ui/consts/miri_unleashed/box.rs

```diff
@@ -11,4 +11,7 @@ static TEST_BAD: &mut i32 = {
     //~^ WARN skipping const check
     //~| ERROR could not evaluate static initializer
     //~| NOTE heap allocations
+    //~| WARN skipping const checks
+    //~| WARN skipping const checks
+    //~| WARN skipping const checks
```

Comment:
Okay this is getting a bit ridiculous, I wonder if we can deduplicate better? Maybe use `tcx.sess.warn` without a span?

Reply:
I think if you remove all `WARN` and `NOTE` from a test, it'll not require any, only errors are strictly required to be annotated

Reply:
But yea, deduplicating warnings like we do lints makes sense. I don't think we should warn without spans, that'll be fairly useless, then we can just not do it at all

Reply:
I want the `NOTE` though to make sure the error message makes sense.

> deduplicating warnings like we do lints makes sense

How'd that work here?

Reply:
We have https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/struct.HandlerInner.html#structfield.emitted_diagnostics and the reason it doesn't work is because we don't do it for tests. So... you need to add `compile-flags: -Z deduplicate-diagnostics=yes` to the tests where you want to dedup errors/warnings.

Reply:
The spans are different though, so this will not help much.

Reply:
Okay, one of the 4 messages for this const got removed... that's not helpful.

Maybe I can somehow use the span of the const, not the oepration inside the const, which should help deduplicate more?

Reply:
All right, the combination of using the item's span and `-Zdeduplicate-diagnostics` helps.